metrics/exporter.go

```diff
@@ -183,6 +183,10 @@ func isNewExporterRequired(newConfig *metricsConfig) bool {
 		return newConfig.collectorAddress != cc.collectorAddress || newConfig.requireSecure != cc.requireSecure
 	}
 
+	if newConfig.backendDestination == prometheus {
+		return newConfig.prometheusHost != cc.prometheusHost || newConfig.prometheusPort != cc.prometheusPort
+	}
+
```

Comment:
I am wondering why we didn't have this here before?

Reply:
I wonder if all the effort to avoid reloading the config is too clever by half, but this seems like a good fix for now.